panic in Extract() when capture group doesn't participate in match

The `Matcher.Extract()` method panics with `slice bounds out of range [:-1]` when a regex pattern contains optional or alternate capture groups that don't participate in the match.

### Steps to Reproduce
```go
package main

import (
	"fmt"
	"github.com/gijsbers/go-pcre"
)

func main() {
	// Pattern with optional group
	re := pcre.MustCompile(`foo(bar)?baz`, 0)
	matcher := re.NewMatcher()
	
	// Match string where optional group doesn't participate
	matcher.MatchString("foobaz", 0)
	
	// This panics
	groups := matcher.Extract()
	fmt.Println(groups)
}
```
### Expected Behavior

- `Extract()` should return `nil` for capture groups that didn't participate in the match
- No panic should occur

### Actual Behavior
```
panic: runtime error: slice bounds out of range [:-1]
goroutine 1 [running]:
github.com/gijsbers/go-pcre.(*Matcher).Extract(...)
    /go/pkg/mod/github.com/gijsbers/go-pcre@v1.0.0/pcre.go:476
```

--- a/pcre.go
+++ b/pcre.go
@@ -473,7 +473,9 @@ func (m *Matcher) Extract() [][]byte {
 	for i := 1; i <= m.groups; i++ {
 		x0 := m.ovector[2*i]
 		x1 := m.ovector[2*i+1]
-		extract[i] = m.subjectb[x0:x1]
+		if x0 >= 0 && x1 >= 0 {
+			extract[i] = m.subjectb[x0:x1]
+		}
 	}
 	return extract
 }

